DQM/L1TMonitor/src/L1TStage2Shower.cc

```diff
@@ -0,0 +1,183 @@
+    cscShowerOccupancyLoose->setBinLabel(ybin, "ME-" + suffix_label[ybin - 1], 2);
+    cscShowerOccupancyLoose->setBinLabel(19 - ybin, "ME+" + suffix_label[ybin - 1], 2);
+  }
+
+  cscShowerOccupancyNom = ibooker.book2D("cscShowerOccupancyNom", "CSC Nominal Shower Occupancy", 36, 1, 37, 18, 0, 18);
+  cscShowerOccupancyNom->setAxisTitle("Chamber", 1);
+  for (int ybin = 1; ybin <= 9; ++ybin) {
+    cscShowerOccupancyNom->setBinLabel(ybin, "ME-" + suffix_label[ybin - 1], 2);
+    cscShowerOccupancyNom->setBinLabel(19 - ybin, "ME+" + suffix_label[ybin - 1], 2);
+  }
+
+  cscShowerOccupancyTight =
+      ibooker.book2D("cscShowerOccupancyTight", "CSC Tight Shower Occupancy", 36, 1, 37, 18, 0, 18);
+  cscShowerOccupancyTight->setAxisTitle("Chamber", 1);
+  for (int ybin = 1; ybin <= 9; ++ybin) {
+    cscShowerOccupancyTight->setBinLabel(ybin, "ME-" + suffix_label[ybin - 1], 2);
+    cscShowerOccupancyTight->setBinLabel(19 - ybin, "ME+" + suffix_label[ybin - 1], 2);
+  }
+
+  cscShowerStationRing =
+      ibooker.book2D("cscShowerStationRing", "CSC shower types in stations and rings", 4, 0, 4, 18, 0, 18);
+  cscShowerStationRing->setAxisTitle("Type", 1);
+  cscShowerStationRing->setBinLabel(1, "Total", 1);
+  cscShowerStationRing->setBinLabel(2, "Loose", 1);
+  cscShowerStationRing->setBinLabel(3, "Nominal", 1);
+  cscShowerStationRing->setBinLabel(4, "Tight", 1);
+  for (int ybin = 1; ybin <= 9; ++ybin) {
+    cscShowerStationRing->setBinLabel(ybin, "ME-" + suffix_label[ybin - 1], 2);
+    cscShowerStationRing->setBinLabel(19 - ybin, "ME+" + suffix_label[ybin - 1], 2);
+  }
+
+  cscShowerChamber = ibooker.book2D("cscShowerChamber", "CSC shower types in chambers", 36, 1, 37, 4, 0, 4);
+  cscShowerChamber->setAxisTitle("Chamber", 1);
+  cscShowerChamber->setAxisTitle("Type", 2);
+  cscShowerChamber->setBinLabel(1, "Total", 2);
+  cscShowerChamber->setBinLabel(2, "Loose", 2);
+  cscShowerChamber->setBinLabel(3, "Nominal", 2);
+  cscShowerChamber->setBinLabel(4, "Tight", 2);
+
+  // EMTF regional shower
+  emtfShowerTypeOccupancy = ibooker.book2D("emtfShowerTypeOccupancy", "EMTF shower Type Occupancy", 6, 1, 7, 8, 0, 8);
+  emtfShowerTypeOccupancy->setAxisTitle("Sector", 1);
+  emtfShowerTypeOccupancy->setBinLabel(8, "ME+ Tight", 2);
+  emtfShowerTypeOccupancy->setBinLabel(7, "ME+ 2Loose", 2);
+  emtfShowerTypeOccupancy->setBinLabel(6, "ME+ Nom", 2);
+  emtfShowerTypeOccupancy->setBinLabel(5, "ME+ Tot", 2);
+  emtfShowerTypeOccupancy->setBinLabel(4, "ME- Tot", 2);
+  emtfShowerTypeOccupancy->setBinLabel(3, "ME- Nom", 2);
+  emtfShowerTypeOccupancy->setBinLabel(2, "ME- 2Loose", 2);
+  emtfShowerTypeOccupancy->setBinLabel(1, "ME- Tight", 2);
+}
+
+void L1TStage2Shower::analyze(const edm::Event& e, const edm::EventSetup& c) {
+  edm::Handle<l1t::RegionalMuonShowerBxCollection> EmtfShowers;
+  e.getByToken(EMTFShowerToken, EmtfShowers);
+
+  edm::Handle<CSCShowerDigiCollection> CscShowers;
+  e.getByToken(CSCShowerToken, CscShowers);
```

Comment:
If you don't need the Handle's then it is faster:
```suggestion
  l1t::RegionalMuonShowerBxCollection const& EmtfShowers = e.get(EMTFShowerToken);
  CSCShowerDigiCollection const& CscShowers = e.get(CSCShowerToken);
```